File: todolist/src/types/meal.ts
export interface MealPayment {
  cost: number;
  account_id: string;
  txn_id: string;
}

export interface MealPaymentInput {
  cost: number;
  account_id: string;
}

export interface DailyMeal {
  date: string;
  breakfast: string;
  lunch: string;
  dinner: string;
  drinks: string;
  breakfast_cost: number;
  breakfast_account_id: string;
  breakfast_txn_id: string;
  lunch_cost: number;
  lunch_account_id: string;
  lunch_txn_id: string;
  dinner_cost: number;
  dinner_account_id: string;
  dinner_txn_id: string;
  drinks_cost: number;
  drinks_account_id: string;
  drinks_txn_id: string;
  created_at: string;
  updated_at: string;
}

export interface SaveMealParams {
  date: string;
  breakfast: string;
  lunch: string;
  dinner: string;
  drinks: string;
  breakfast_payment?: MealPaymentInput;
  lunch_payment?: MealPaymentInput;
  dinner_payment?: MealPaymentInput;
  drinks_payment?: MealPaymentInput;
}

File: todolist/src-tauri/src/models/meal.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct DailyMeal {
    pub date: String,
    pub breakfast: String,
    pub lunch: String,
    pub dinner: String,
    pub drinks: String,
    pub breakfast_cost: f64,
    pub breakfast_account_id: String,
    pub breakfast_txn_id: String,
    pub lunch_cost: f64,
    pub lunch_account_id: String,
    pub lunch_txn_id: String,
    pub dinner_cost: f64,
    pub dinner_account_id: String,
    pub dinner_txn_id: String,
    pub drinks_cost: f64,
    pub drinks_account_id: String,
    pub drinks_txn_id: String,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Deserialize)]
pub struct MealPaymentInput {
    #[serde(default)]
    pub cost: f64,
    #[serde(default)]
    pub account_id: String,
}

#[derive(Debug, Deserialize)]
pub struct SaveMealParams {
    pub date: String,
    pub breakfast: String,
    pub lunch: String,
    pub dinner: String,
    pub drinks: String,
    #[serde(default)]
    pub breakfast_payment: Option<MealPaymentInput>,
    #[serde(default)]
    pub lunch_payment: Option<MealPaymentInput>,
    #[serde(default)]
    pub dinner_payment: Option<MealPaymentInput>,
    #[serde(default)]
    pub drinks_payment: Option<MealPaymentInput>,
}
